contactaccessrights/tests/phpunit/CRM/Contactaccessrights/Utils/ACLTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobContract as JobContractFabricator;
+use CRM_Contactaccessrights_Test_Fabricator_Rights as RightsFabricator;
+
+/**
+ * Class CRM_Contactaccessrights_BAO_RightsTest
+ * Tests Rights BAO class for contact's access rights to civicrm.
+ * 
+ * @group headless
+ * @author Camilo Rodriguez - camilo@compucorp.co.uk
+ */
+class CRM_Contactaccessrights_Utils_ACLTest extends PHPUnit_Framework_TestCase implements
+  HeadlessInterface, TransactionalInterface {
+
+	/**
+	 * Installs extensions required for test
+	 * 
+	 * @return \Civi\Test\CiviEnvBuilder
+	 */
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->install('uk.co.compucorp.civicrm.hrcore')
+      ->install('org.civicrm.hrjobcontract')
+      ->installMe(__DIR__)
+      ->apply();
+  }
+  
+	/**
+	 * Tests if ACL Utility class is building the Job Roles Clause appropriately,
+	 * using option_value.value for Locations and Regions in the table join.
+	 */
+  public function testJobRolesClauseWhereACL() {
+    // Setup local admin
+    $localAdmin = ContactFabricator::fabricate();
+		
+		// Create Regions and Locations
+		$region1 = $this->createOpValue('hrjc_region', 'Region 1');

```

Comment:
There are fabricators for option values and option groups. You should use them instead